### PR TITLE
Update to Gulp v4 and major refactoring

### PR DESCRIPTION
**Pull request by thomasmattheussen**
- Update dependencies
- Update to gulp v4 and refactor tasks due to breaking changes
- Update documentation
- Removed production task in favor of unified approach through `$ gulp` and `$ gulp --production`
- Removed some legacy comments and documentation

### CODE REVIEW

File: gulpfile.js/lib/isProductionBuild.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+const argv = require('minimist')(process.argv.slice(2));
+
+module.exports = () => {
```
Comment (samueljoos) [inline gulpfile.js/lib/isProductionBuild.js:3]: Seems a bit useless to put this in a function. Although it won't impact performance much

**Review by samueljoos — APPROVED** [review]
Tested it and everything seems to work.

Although, I have 2 thoughts:
1. shouldn't we add "babel-polyfill" with the option useBuiltIns: 'usage'. This way we will always have the necessary pollyfill's for our projects. 
2. recipe-react could use a little update, although it has nothing to do with your current changes but with the changes to the babelrc shizzle. 

These thoughts can be fixed in an other feature so this one is approved!

File: gulpfile.js/lib/isProductionBuild.js (hunk 1 above)
Comment (thomasmattheussen) [reply gulpfile.js/lib/isProductionBuild.js:3]: Hmm yeah, maybe... Can't really find a good argument to do this except: 'it feels better to me' 😅

**Comment by thomasmattheussen** [comment]
Good suggestions, we will look into those in a separate issue.

**Pull request merged**